test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -577,6 +581,79 @@ def __set_test_end_data(self, test, err=None):
         test.captured_logs = self.result.handler.get_logs()
 
+    def __get_param_dictionary(self):
+        """
+        Method to convert data in param into dictionary of counts
+        """
+        paramkeys = ['server', 'servers', 'mom', 'moms', 'comms', 'client']
+        tparam_dic = {}
+        tparam_contents = {}
+        for key in paramkeys:
+            tparam_contents[key] = []
+        tparam_dic.update(default_requirements)
+        if self.param is None:
+            return tparam_dic
+        for h in self.param.split(','):
+            if '=' in h:
+                k, v = h.split('=')
+                if k in paramkeys:
+                    tparam_contents[k] = [v.replace(':', ',')]
+                    if (k == 'server' or k == 'servers'):
+                        tparam_dic['num_servers'] = len(v.split(':'))
+                    if (k == 'mom' or k == 'moms'):
+                        tparam_dic['num_moms'] = len(v.split(':'))
+                    if k == 'comms':
+                        tparam_dic['num_comms'] = len(v.split(':'))
+                    if k == 'clients':
+                        tparam_dic['num_clients'] = len(v.split(':'))
+        for pkey in paramkeys:
+            if pkey in ['server', 'mom', 'comms', 'client']:
+                if not tparam_contents[pkey]:
+                    tparam_contents[pkey] = [platform.node().split('.', 1)[0]]
+        if not (
+            (set(tparam_contents['servers']) & set(tparam_contents['moms'])) or
+            (set(tparam_contents['server']) & set(tparam_contents['moms'])) or
+            (set(tparam_contents['servers']) & set(tparam_contents['mom'])) or
+            (set(tparam_contents['server']) & set(tparam_contents['mom']))
+        ):
+            tparam_dic['no_mom_on_server'] = True
+        if not (
+            (set(tparam_contents['servers']) &
+             set(tparam_contents['comms'])) or
+            (set(tparam_contents['server']) & set(tparam_contents['comms'])) or
+            (set(tparam_contents['moms']) & set(tparam_contents['comms'])) or
+            (set(tparam_contents['mom']) & set(tparam_contents['comms']))
+        ):
+            tparam_dic['no_comm_on_server'] = True
+        if (
+            (set(tparam_contents['moms']) & set(tparam_contents['comms'])) or
+            (set(tparam_contents['mom']) & set(tparam_contents['comms']))
+        ):
+            tparam_dic['no_comm_on_mom'] = False
+        return tparam_dic
+
+    @staticmethod
+    def __are_requirements_matching(param_count=None, test=None):
+        """
+        Validates test requirements against test cluster information
+        returns True on match or False otherwise
+        """
+        keylist = ['num_servers', 'num_moms', 'num_comms', 'num_clients']
+        flaglist = ['no_mom_on_server', 'no_comm_on_server', 'no_comm_on_mom']
+        if test is not None:
+            method = getattr(test.test, getattr(test.test, '_testMethodName'))
```

Comment:
To be safer side, add default value in getattr and put if condition after this line